### PR TITLE
`Development`: Update client tests documentation

### PR DESCRIPTION
**Pull request by JohannesWt**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform--9913.org.readthedocs.build/en/9913/dev/guidelines/client-tests.html).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

Regarding the Client Migration Process the client test documentation should also be updated.

### Description

This PR updates the client tests documentation to the newest Angular standards.
[Updated documentation page](https://artemis-platform--9913.org.readthedocs.build/en/9913/dev/guidelines/client-tests.html)

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

## Summary by CodeRabbit

- **Documentation**
	- Updated guidelines for client testing practices using Jest in Angular.
	- Enhanced clarity on asynchronous module configuration and best practices for component isolation.
	- Revised examples illustrating good and bad testing practices.
	- Updated HTTP testing recommendations to align with the latest Angular standards.
	- Expanded guidelines on naming conventions and verification functions for improved error messaging.

### CODE REVIEW

File: docs/dev/guidelines/client-tests.rst [hunk 1]
```diff
@@ -19,9 +19,8 @@ The most basic test looks similar to this:
         let someComponent: SomeComponent;
 
         beforeEach(() => {
```
Comment (b-fein) [inline docs/dev/guidelines/client-tests.rst:21]:
```suggestion
        beforeEach(async () => {
```
I guess this is required now due to the `await TestBed.…`?

File: docs/dev/guidelines/client-tests.rst [hunk 2]
```diff
@@ -61,23 +59,24 @@ Some guidelines:
             ...
 
             beforeEach(() => {
```
Comment (b-fein) [inline docs/dev/guidelines/client-tests.rst:61]:
```suggestion
            beforeEach(async () => {
```

File: docs/dev/guidelines/client-tests.rst [hunk 3]
```diff
@@ -95,9 +94,11 @@ Some guidelines:
                 ...
 
                 beforeEach(() => {
```
Comment (b-fein) [inline docs/dev/guidelines/client-tests.rst:96]:
```suggestion
                beforeEach(async () => {
```

File: docs/dev/guidelines/client-tests.rst [hunk 4]
```diff
@@ -222,20 +225,17 @@ Some guidelines:
             let someComponent: SomeComponent;
 
             beforeEach(() => {
```
Comment (b-fein) [inline docs/dev/guidelines/client-tests.rst:227]:
```suggestion
            beforeEach(async () => {
```

**Review by az108 — APPROVED** [review]
updated documentation LGTM 👍

**Review by b-fein — APPROVED** [review]

**Review by HawKhiem — APPROVED** [review]
LGTM changes. I would also want to recommend adding a table of content for better clarity

**Review by krusche — APPROVED** [review]

**Pull request merged**